neqo-transport/src/crypto.rs

```diff
@@ -84,6 +84,17 @@ impl Crypto {
         });
     }
 
+    /// Find the highest state that has keys.
+    pub fn highest_state(&self) -> Epoch {
+        let mut val = 0;
+        for (idx, state) in self.states.iter().enumerate() {
```

Comment:
I had a similar patch, but simply this for loop did not work. I needed to call crypto.obtain_crypto_state to actually initialize all the states. Can you write some tests and see if it is working properly?

Reply:
My take was that the goal was to "send CLOSE frames in all connections we have keys for". If so, we only want to find existing keys and not initialize more keys. Since we never currently drop keys, this will mean sending in spaces 0->highest.

This isn't working for you? If I run:
`RUST_LOG=neqo_transport::dump ./target/debug/neqo-client https://blog.cloudflare.com/ --omit-read-data` I see three ConnectionClose frames being sent.

I completely agree about some tests. I need to figure out a way to decrypt the three packets in the datagram and verify they each have a close frame. It would also be good to test closing during initial and during handshake, and seeing the right thing happen.